modules/ort/src/main/java/org/eclipse/sw360/antenna/ort/resolver/OrtResultArtifactResolver.java

```diff
@@ -0,0 +1,154 @@
+            default:
+                return Optional.of(mapSimpleCoordinates(name, version));
+        }
+    }
+
+    private static ArtifactCoordinates mapDotNetCoordinates(String name, String version) {
+        return new DotNetCoordinates.DotNetCoordinatesBuilder()
+                .setPackageId(name)
+                .setVersion(version)
+                .build();
+    }
+
+    private static ArtifactCoordinates mapMavenCoordinates(String namespace, String name, String version) {
+        return new MavenCoordinates.MavenCoordinatesBuilder()
+                .setGroupId(namespace)
+                .setVersion(version)
+                .setArtifactId(name)
+                .build();
+    }
+
+    private static ArtifactCoordinates mapJavaScriptCoordinates(String name, String version) {
+        return new JavaScriptCoordinates.JavaScriptCoordinatesBuilder()
+                .setName(name)
+                .setVersion(version)
+                .setArtifactId(name + "-" + version)
+                .build();
+    }
+
+    private static ArtifactCoordinates mapSimpleCoordinates(String name, String version) {
+        return new GenericArtifactCoordinates(name, version);
+    }
+
+    private static Optional<ArtifactSourceUrl> mapSourceUrl(Package pkg) {
+        final Stream<String> urls = Stream.of(pkg.getVcsProcessed().getUrl(), pkg.getSourceArtifact().getUrl());
+
+        return urls.filter(u -> !u.isEmpty())
+                .findFirst()
+                .map(ArtifactSourceUrl::new);
+    }
+
+    private static Optional<DeclaredLicenseInformation> mapDeclaredLicense(Package pkg) {
+        return Optional.of(pkg.getDeclaredLicenses())
+                .map(LicenseSupport::mapLicenses)
+                .map(DeclaredLicenseInformation::new);
+    }
+
+    private static Optional<ArtifactFilename> mapFilename(Package pkg) {
+        final Stream<RemoteArtifact> artifacts = Stream.of(pkg.getSourceArtifact(), pkg.getBinaryArtifact());
+
+        return artifacts
+                .filter(a -> !a.getUrl().isEmpty())
+                .findFirst()
+                .map(a -> {
+                    String fileName = a.getUrl();
+                    String hash = a.getHash().getValue();
+                    String hashAlgorithm = a.getHash().getAlgorithm().toString();
+                    return new ArtifactFilename(fileName, hash, hashAlgorithm);
+                });
+    }
+
+    private static Optional<ArtifactHomepage> mapHomepage(Package pkg) {
+        return Optional.of(pkg.getHomepageUrl())
+                .map(ArtifactHomepage::new);
+    }
+
+    private Optional<ObservedLicenseInformation> mapObservedLicense(Package pkg) {
+        Collection<String> licenses = Optional.ofNullable(licenseFindings.get(pkg.getId()))
+                .map(Map::keySet)
+                .orElse(Collections.emptySet()).stream()
+                .map(LicenseFinding::getLicense)
+                .collect(Collectors.toList());
+
```

Comment:
Why are declared and observed licenses handled differently?

Reply:
Both are valid questions, but I'd like to address them separately. This PR is only about porting the existing code (which already had the issues you pointed out) to use the ORT model directly. I.e. no change of functionality is done, just porting to another API.

For reference, see to original code at

* https://github.com/eclipse/antenna/pull/247/files#diff-9220c0f0beaa323e93534dc8c4cf5c56L68

* https://github.com/eclipse/antenna/pull/247/files#diff-5363d56ff7d691e5d496f9256fd61b73L68

Reply:
Differently in what way?

Reply:
Compared to my comment above, the handling of the observed information is much more difficult, and not easy to understand.

Reply:
The ORT data structures for detected (aka observed) licenses are designed with associating them with copyright statements in mind. That's why a lookup by artifact is rather convoluted compared to the lookup of a declared license.